p2p/src/authenticated/discovery/actors/spawner/actor.rs

```diff
@@ -132,9 +132,18 @@ impl<E: Spawner + Clock + Rng + CryptoRng + Metrics, O: Sink, I: Stream, C: Publ
                             let is_dialer = matches!(reservation.metadata(), Metadata::Dialer(..));
                             let info_verifier = self.info_verifier.clone();
                             move |context| async move {
+                                // Get greeting from tracker (returns None if not eligible)
+                                let Some(greeting) = tracker.connect(peer.clone(), is_dialer).await
```

Comment:
This new approach avoids spinning up a peer if we connect at the same time via some other means